### PR TITLE
Clean up critical sonar smells

### PR DESCRIPTION
**Pull request by rottebds**

### CODE REVIEW

File: hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/workflow/codelocation/CodeLocationNameManager.java [hunk 1]
```diff
@@ -34,22 +34,14 @@
 
 public class CodeLocationNameManager {
     private final DetectConfigWrapper detectConfigWrapper;
-    private final BomCodeLocationNameService bomCodeLocationNameService;
-    private final DockerCodeLocationNameService dockerCodeLocationNameService;
-    private final ScanCodeLocationNameService scanCodeLocationNameService;
-    private final DockerScanCodeLocationNameService dockerScanCodeLocationNameService;
+    private final CodeLocationNameService codeLocationNameFactory;
```
Comment (taikuukaits) [inline hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/workflow/codelocation/CodeLocationNameManager.java]: Factory or Service?

File: hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/workflow/codelocation/CodeLocationNameService.java [hunk 2]
```diff
@@ -23,24 +23,144 @@
...
+    public static final int MAXIMUM_CODE_LOCATION_NAME_LENGTH = 250;
     private final DetectFileFinder detectFileFinder;
 
     public CodeLocationNameService(final DetectFileFinder detectFileFinder) {
         this.detectFileFinder = detectFileFinder;
     }
 
-    public DetectFileFinder getDetectFileFinder() {
-        return detectFileFinder;
+    public String createBomCodeLocationName(final String detectSourcePath, final String sourcePath, final ExternalId externalId, final BomToolGroupType bomToolType, final String prefix, final String suffix) {
```
Comment (jamesrichard91) [inline hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/workflow/codelocation/CodeLocationNameService.java]: Is there a reason, you collapsed the services into this one class? Or do you just prefer this approach?

File: hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/util/DetectFileFinder.java [hunk 3]
```diff
@@ -92,33 +69,27 @@ public boolean containsAllFilesToDepth(final String sourcePath, final int maxDep
...
-            return null;
-        } else if (foundFiles.size() > 1) {
-            logger.debug(String.format("Found multiple matches for %s in %s", filenamePattern, sourceDirectory.getAbsolutePath()));
-            logger.debug(String.format("Using %s", foundFiles.get(0)));
+            foundFile = null;
+        } else {
+            foundFile = foundFiles.get(0);
+            if (foundFiles.size() > 1) {
+                logger.debug(String.format("Found multiple matches for %s in %s", filenamePattern, sourceDirectory.getAbsolutePath()));
+                logger.debug(String.format("Using %s", foundFile));
```
Comment (taikuukaits) [inline hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/util/DetectFileFinder.java]: Are we sure we want to return a file if we found multiple matches? Seems potentially unpredictable.

Comment (taikuukaits) [reply hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/util/DetectFileFinder.java]: I realize the old one did the same thing so it's probably fine.

**Pull request merged**